js/extension/features/editing-notepad.js

```diff
@@ -0,0 +1,384 @@
+  }
+
+  const isNotepadInitiallyVisible = getNotepadVisiblePreference() === null ? false : true;
+
+  const initialNotepadClasses = isNotepadInitiallyVisible ? '' : hiddenClass;
+  const initialToggleButtonClasses = isNotepadInitiallyVisible ? rotateClass : '';
+  const initialTitle = isNotepadInitiallyVisible ? hideTitle : showTitle;
+
+  const markup = `
+    <div id="${notepadEditCoverId}"></div>
+    <div id="${notepadId}" class="${initialNotepadClasses}">
+      <div>
+        <div class="${buttonsWrapperClass}">
+          <button id="${toggleButtonId}" class="${initialToggleButtonClasses} ${buttonClass} button" title="${initialTitle}">◀</button>
+          <button id="${editButtonId}" class="${buttonClass} button" title="${editTitle}">${editEmoji}</button>
+        </div>
+        <div id="${areaWrapperId}">
+          <p id="${notepadAreaId}" contenteditable="false" tabindex="-1"></p>
+        </div>
+      </div>
+    </div>
+  `;
+
+  const css = `
+
+    #${notepadEditCoverId} {
+      position: fixed;
+      top: 0;
+      bottom: 0;
+      right: 0;
+      left: 0;
+      visibility: hidden;
+      display: none;
+      z-index: 0;
+    }
+
+    #${notepadEditCoverId}.${visibleClass} {
+      visibility: visible;
+      display: unset;
+    }
+
+    #${notepadId} {
+      position: fixed;
+      right: 0;
+      top: 50%;
+      transform: translateY(-50%);
+      height: 700px;
+      width: 500px;
+      min-width: 500px;
+      max-width: 500px;
+      transition: all 0.3s ease-in-out;
+      z-index: 1;
+    }
+
+    #${notepadId}.${hiddenClass} {
+      right: -501px;
+    }
+
+    #${notepadId} > div {
+      height: 100%;
+      width: 100%;
+      min-width: 500px;
+      max-width: 500px;
+      display: flex;
+      flex-flow: row nowrap;
+      justify-content: flex-end;
+      align-items: center;
+      padding: 0;
+      margin: 0;
+    }
+
+    #${areaWrapperId} {
+      width: 100%;
+      height: 100%;
+      min-width: 500px;
+      max-width: 500px;
+      padding: 0;
+      margin: 0;
+      z-index: 1;
+    }
+
+    #${notepadAreaId} {
+      height: 100%;
+      width: 100%;
+      min-width: 500px;
+      max-width: 500px;
+      margin: 0;
+      background-color: white;
+      outline: 1px solid grey !important;
+      opacity: 0.7;
+      transition: opacity 0.3s ease-in-out;
+      overflow-y: auto;
+      overflow-x: hidden;
+      overflow-wrap: break-word;
+    }
+
+    .de-dark-theme #${notepadAreaId} {
+      background-color: #292929 !important;
+    }
+
+    #${notepadAreaId} > *:first-child {
+      padding-top: 5px;
+    }
+
+    #${notepadAreaId} > *:last-child {
+      padding-bottom: 5px;
+    }
+
+    #${notepadAreaId} > * {
+      padding-left: 5px;
+    }
+
+    .${notepadFocusedClass} #${notepadAreaId} {
+      opacity: 1;
+    }
+
+    .${notepadEditingClass} #${notepadAreaId} {
+      cursor: text;
+      outline: 1px solid red !important;
+      box-shadow: 0 0 10px red !important;
```

Comment:
I'm hesitant to use `red` while editing since `red` is (IMO) generally used to indicate an error. Do you have any thoughts on using a different color or approach when indicating the editing state?

Reply:
That's true.  I just went with red to start with and ended up not changing it.  Could go with a `goldenrod`?

Here's what that would look like:

![image](https://user-images.githubusercontent.com/4356159/101859537-e240bc00-3b39-11eb-84b5-6390d6c62545.png)

Reply:
Yeah, works for me 👍